packages/cli-e2e/__tests__/orgResources.specs.ts

```diff
@@ -94,25 +126,66 @@ describe('org:resources', () => {
   });
 
   describe('org:resources:preview', () => {
-    it(
-      'should preview the snapshot',
-      async () => {
-        const previewTerminal = previewChange(testOrgId, processManager);
-
-        const expectedOutput = [
-          'Extensions',
-          '\\+   1 to create',
-          'Fields',
-          '\\+   2 to create',
-          'Filters',
-          '\\+   1 to create',
-        ].join('\\s*');
-        const regex = new RegExp(expectedOutput, 'gm');
-
-        await previewTerminal.when(regex).on('stdout').do().once();
-      },
-      defaultTimeout
-    );
+    describe('when resources are synchronized', () => {
+      it(
+        'should preview the snapshot',
+        async () => {
+          const previewTerminal = previewChange(
+            testOrgId,
+            processManager,
+            'org-config-preview-sync'
+          );
+
+          const expectedOutput = [
+            'Extensions',
+            '\\+   1 to create',
+            'Fields',
+            '\\+   2 to create',
+            'Filters',
+            '\\+   1 to create',
+            'Ml models',
+            '\\+   1 to create',
+          ].join('\\s*');
+          const regex = new RegExp(expectedOutput, 'gm');
+
+          await previewTerminal.when(regex).on('stdout').do().once();
+        },
+        defaultTimeout
+      );
+    });
+
+    describe('when resources are not synchronized', () => {
+      beforeAll(async () => {
+        await createFieldWithoutUsingSnapshot(platformClient);
+      });
+
+      it(
+        'should throw a synchronization warning on a field',
+        async () => {
+          const previewTerminal = previewChange(
+            testOrgId,
+            processManager,
+            'org-config-preview-unsync'
+          );
+          const warningRegex = new RegExp(
+            /Synchronization plan matched all resources, do you want to proceed/
+          );
+
+          const previewTerminalExitPromise = previewTerminal
+            .when(/Previewing resource changes/)
+            .on('stdout')
+            .do()
+            .once();
+
+          await previewTerminal
+            .when(warningRegex)
+            .on('stderr')
+            .do(answerPrompt(`y${EOL}`))
+            .until(previewTerminalExitPromise);
```

Comment:
There's no 'expectation' in this test 🤔 

The test should fail (nicely if possible and with a nice explanation) when the previewTerminal concludes without going thru the warningRegex.

Here, it won't I reckon.